File: freecodecamp/daily-code-challenge/2026-03-20.js
function getShadow(time) {
  let [h, m] = time.split(":").map(Number);
  let t = h + m / 60;

  if (t < 6 || t >= 18 || t === 12) return "No shadow";

  let diff = Math.abs(12 - t);
  let length = diff ** 3;

  let direction = t < 12 ? "west" : "east";

  return `${length}ft ${direction}`;
}

File: freecodecamp/daily-code-challenge/2026-03-20.py
def get_shadow(time):
    h, m = map(int, time.split(":"))
    t = h + m / 60

    if t < 6 or t >= 18 or t == 12:
        return "No shadow"

    diff = abs(12 - t)
    length = diff ** 3

    if length.is_integer():
        length = int(length)

    direction = "west" if t < 12 else "east"

    return f"{length}ft {direction}"